src/handler/archive.ts

```diff
@@ -0,0 +1,78 @@
+import 'dotenv/config';
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import logger from '../util/logger';
+import { validateArchiveErrors } from '../validators/archive';
+import { addHttpHeaders } from '../util/httpHeaders';
+import { archiveRecord, getBySystemNumberAndCreatedTimestamp } from '../services/database';
+import { getUserDetails } from '../services/user';
+import { formatTechRecord } from '../util/formatTechRecord';
+import { ArchiveRecord, ArchiveRecordRequestBody } from '../models/archive';
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Archive end point called');
+
+  const archiveErrors = validateArchiveErrors(event);
+
+  if (archiveErrors) {
+    return addHttpHeaders(archiveErrors);
+  }
+
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const body: ArchiveRecordRequestBody = await JSON.parse(event.body as string);
+
+    if (!body.reasonForArchiving) {
+      return {
+        statusCode: 400,
+        body: 'Reason for archiving not provided',
+      };
+    }
+
+    const systemNumber = event?.pathParameters?.systemNumber as string;
+    const createdTimestamp = event?.pathParameters?.createdTimestamp as string;
+    const userDetails = getUserDetails(event.headers.Authorization as string);
+
+    logger.debug(`Get from database with sysNum ${systemNumber} and timestamp ${createdTimestamp}`);
+
+    const record = await getBySystemNumberAndCreatedTimestamp(systemNumber, createdTimestamp) as ArchiveRecord;
+
+    logger.debug(`result is: ${JSON.stringify(record)}`);
+
+    if (!record || !Object.keys(record).length) {
+      return addHttpHeaders({
+        statusCode: 404,
+        body: `No record found matching sysNum ${systemNumber} and timestamp ${createdTimestamp}`,
+      });
+    }
+
+    if (record.techRecord_statusCode === 'archived') {
+      return addHttpHeaders({
+        statusCode: 400,
+        body: 'Cannot archive an archived record',
+      });
+    }
```

Comment:
Similar to the amend vin, could we not just return a `200` if the record is already archived ?

Reply:
I think logically, it hasn't actually archived anything so I think it might be wrong to return 200 - maybe? I think the 400 response suits this situation where it's tried to do it, but hasn't been successfu